x/ccv/consumer/keeper/changeover_test.go

```diff
@@ -0,0 +1,162 @@
+package keeper_test
+
+import (
+	"testing"
+
+	sdkcryptocodec "github.com/cosmos/cosmos-sdk/crypto/codec"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+	"github.com/cosmos/interchain-security/testutil/crypto"
+	uthelpers "github.com/cosmos/interchain-security/testutil/keeper"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/require"
+	abci "github.com/tendermint/tendermint/abci/types"
+)
+
+func TestChangeoverToConsumer(t *testing.T) {
+
+	cIds := []crypto.CryptoIdentity{}
+	for i := 0; i < 10; i++ {
+		cIds = append(cIds, *crypto.NewCryptoIdentityFromIntSeed(i + 42834729))
+	}
+
+	// Instantiate 5 sov validators for use in test
+	sovVals := []stakingtypes.Validator{
+		cIds[0].SDKStakingValidator(),
+		cIds[1].SDKStakingValidator(),
+		cIds[2].SDKStakingValidator(),
+		cIds[3].SDKStakingValidator(),
+		cIds[4].SDKStakingValidator(),
+	}
+
+	// Instantiate 5 ics val updates for use in test
+	initialValUpdates := []abci.ValidatorUpdate{
+		{Power: 55, PubKey: cIds[5].TMProtoCryptoPublicKey()},
+		{Power: 87324, PubKey: cIds[6].TMProtoCryptoPublicKey()},
+		{Power: 2, PubKey: cIds[7].TMProtoCryptoPublicKey()},
+		{Power: 42389479, PubKey: cIds[8].TMProtoCryptoPublicKey()},
+		{Power: 9089080, PubKey: cIds[9].TMProtoCryptoPublicKey()},
+	}
+
+	testCases := []struct {
+		name string
+		// Last standalone validators that will be mock returned from stakingKeeper.GetLastValidators()
+		lastSovVals []stakingtypes.Validator
+		// Val updates corresponding to initial valset set for ccv set initGenesis
+		initialValUpdates []abci.ValidatorUpdate
+		// Expected length of val updates returned from ChangeoverToConsumer()
+		expectedReturnValUpdatesLen int
+	}{
+		{
+			name:                        "no sov vals, no initial val updates",
+			lastSovVals:                 []stakingtypes.Validator{},
+			initialValUpdates:           []abci.ValidatorUpdate{},
+			expectedReturnValUpdatesLen: 0,
+		},
+		{
+			name:                        "one sov val, no initial val updates",
+			lastSovVals:                 []stakingtypes.Validator{sovVals[0]},
+			initialValUpdates:           []abci.ValidatorUpdate{},
+			expectedReturnValUpdatesLen: 1,
+		},
+		{
+			name:                        "no sov vals, one initial val update",
+			lastSovVals:                 []stakingtypes.Validator{},
+			initialValUpdates:           []abci.ValidatorUpdate{initialValUpdates[0]},
+			expectedReturnValUpdatesLen: 1,
+		},
+		{
+			name:                        "one sov val, one initial val update",
+			lastSovVals:                 []stakingtypes.Validator{sovVals[0]},
+			initialValUpdates:           []abci.ValidatorUpdate{initialValUpdates[0]},
+			expectedReturnValUpdatesLen: 2,
+		},
+		{
+			name:                        "five sov vals, five initial val updates",
+			lastSovVals:                 sovVals,
+			initialValUpdates:           initialValUpdates,
+			expectedReturnValUpdatesLen: 10,
+		},
```

Comment:
We can add a test case where a validator is contained in both sov validators and initial valupdates

Reply:
Good call, done in https://github.com/cosmos/interchain-security/commit/f518e9be788380232fe4ddf3a5e56254228279d9